attestation/src/hash.rs

```diff
@@ -1,8 +1,25 @@
 use alloc::string::String;
+use borsh::{BorshDeserialize, BorshSerialize};
 use core::marker::PhantomData;
 use derive_more::{AsRef, Deref, Into};
 
-#[derive(Deref, AsRef, Into)]
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    serde::Serialize,
+    serde::Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+    Deref,
+    AsRef,
+    Into,
+)]
+#[serde(transparent)]
```

Comment:
Note: This is the fix. The rest of the PR is test cases and the refactor to use this Hash type in the contract.